lib/MusicBrainz/LWP.pm

```diff
@@ -0,0 +1,151 @@
+package MusicBrainz::LWP;
+
+use Moose;
+use namespace::autoclean;
+
+use AnyEvent;
+use AnyEvent::HTTP::LWP::UserAgent;
+use DBDefs;
+use HTTP::Status qw( HTTP_SERVICE_UNAVAILABLE );
+use LWP::UserAgent;
+use MooseX::Types::Moose qw( Int );
```

Comment:
I’m curious why is this used called 87 times in the current code, mostly under `lib/MusicBrainz/Server/Edit/`?

Even if it is not actually needed, doesn’t it help a bit with code readability at least?

Reply:
The bareword version `Int` from `MooseX::Types::Moose` can be used to detect typos at compile time (as in mistyping "Int"), for instance, so you might prefer it. But in that case we ought to replace `'Int'` below with `Int` -  if we're going through the trouble of importing the bareword version we may as well use it to get the compile-time benefits. :)

Reply:
As in <https://github.com/metabrainz/musicbrainz-server/pull/1102/commits/9a91190b4f84b80d619a2de37a5733e6db0b6e12>?

Reply:
Yes, except the `LWP::UserAgent` change will probably cause Perl to die because that's not a subroutine (`Int` is).

Reply:
Would that require to define `LwpAgent` like [`DateTime` in `MusicBrainz::Server::Types`](https://github.com/metabrainz/musicbrainz-server/blob/v-2020-06-15/lib/MusicBrainz/Server/Types.pm#L14-L30) otherwise? The trouble might not be worth the compile-time benefit.

Reverted that in https://github.com/metabrainz/musicbrainz-server/pull/1102/commits/f80a38d7d8a50d1703d113d294524bf511e632f5 for now.

Reply:
Yeah, exactly. You can define one if you think it's worth it, but I rarely bother importing builtin Moose types from `MooseX::Types::Moose` either.

Reply:
I counted around 500 occurrences of `isa` builtin Moose type for around 200 custom types. Importing builtin Moose types from `MooseX::Types::Moose` looks like an easy way to get compile-time check, (even tiny) run-time speed improvement, and better readability (for a layman) with just one more line of code (per file, so actually around 500 lines). Am I missing something?

Reply:
Not really, just that after you write 100 Moose classes, it becomes annoying to have to import builtin types. :) The compile-time check also isn't as helpful as you might hope: it doesn't check if the type is *correct*, just that you didn't misspell it. You still have to actually run the code to see if it works, which would have told you the same thing. Perl is still a dynamic language through and through. (I also don't tend to misspell simple types like Int, Bool, etc.; I'm more likely to type the import wrong on first try.)

Reply:
Fair enough, plus I found that coercion doesn’t work for composed types such as `Maybe[Not[A[Good[Idea]]]]` even if it has been defined for each of these types.